yjit/src/asm/arm64/opnd.rs

```diff
@@ -13,9 +13,12 @@ pub struct A64Reg
 
 impl A64Reg {
     pub fn sub_reg(&self, num_bits: u8) -> Self {
-        assert!(num_bits == 32 || num_bits == 64);
         assert!(num_bits <= self.num_bits);
+        self.with_num_bits(num_bits)
+    }
```

Comment:
We might want to get rid of `sub_reg` and `sub_opnd` and replace them with `with_num_bits` everywhere?

Reply:
agreed. done https://github.com/ruby/ruby/pull/6359/commits/5274fcdf7445153788faad73a4614bc296d8cef8